### DISCUSSION TITLE
Golang Processor

### DISCUSSION THREAD

**Discussion by Wuerike** [redpanda-data/connect#2796] [General]
So, time to time I need some small thing that is not that straight forward to achieve, at this moments I usually use AWK or js processors.

But what I always got me questioning is, why dont we have a golang processor? I just don't trust enough in this interpreters running other languages in go, wouldn't be better use go by it self as a processor?

**Comment by mihaitodor** [comment]
Hey @Wuerike 👋 Thank you for the suggestion! I think that should be fairly straightforward to achieve via https://github.com/traefik/yaegi. I imagine the code would be similar to the `javascript` processor. PRs are welcome, but you can also write your own [plugin](https://github.com/redpanda-data/redpanda-connect-plugin-example).

Also, the `wasm` processor should support binaries produced by tinygo: https://github.com/redpanda-data/benthos/tree/82e275d2a9c33a0079d62d6abf709066bd48e2fd/public/wasm/examples/tinygo
